cgrok/grok.go

```diff
@@ -57,12 +65,16 @@ func New() *Grok {
    With other strings there's a risk of getting the wrong result if memory has been freed and reused. */
 func (grok *Grok) gostringn(str *C.char, len C.int) string {
 	ptr := uintptr(unsafe.Pointer(str))
+	grok.stringCacheLock.RLock()
```

Comment:
By removing this read lock, I get another ~10% on the benchmarks. If I understand correctly, we only need this lock while the `stringCache` is filling up for the very first grok, so would it be worth adding a fastpath that checks if we've completed filling the cache?